elastic/kibana
# [Synthetics 1.0] Monitor Details -> Monitor Summary -> Last test run panels

## Summary

Tracked in https://github.com/elastic/kibana/issues/134204.

Adds the **Last test run** and **Last 10 Test Runs** panels on the Monitor Details -> Summary page. Both of the sections are added in one PR as the state is shared between the two UI sections.

- A parent level Monitor Details directory is created which will house the Summary, History and Error related sub directories.
- A copy is added when a monitor has never run and no information about last test run is available for the selected location. (wasn't in design)
- For ping monitors, label-value list is added for **Last test run** instead of table of steps. (wasn't in design)
- The title **Last 10 Test Runs** will switch to **Test Runs** if number of available past test runs are less than 10. (wasn't in design)
- The empty **Status** trail histogram panel is hidden for now.

### How to test
1. Setup Kibana and a Synthetics project  so that you are able to create monitors via UI as well as push project monitors.
2. Create at least one browser and one lightweight monitor (HTTP, TCP or ICMP monitor).
3. Push at least one browser monitor via @elastic/synthetics npm package.
4. Wait a bit for the results to be ingested.
5. Go to Synthetics -> Monitors -> _click the ... menu of a monitor_ -> Go to monitor.
6. Confirm that you can see expected data (as designs suggest) under **Last test run** and **Last 10 Test Runs**.
7. Repeat steps 5 and 6 for other monitors.
8. Bonus 1 - Use a private location to test the same steps.
9. Bonus 2 - Push a Lightweight Project Monitor using a yaml as Lightweight Project Monitors are a new implementation. Consult the respective PR for instructions if you are up to it.

![Screenshot 2022-07-25 at 17 35 46](https://user-images.githubusercontent.com/2748376/180826787-3c3ec926-7952-431d-a76a-d10544c728f2.png)
![Screenshot 2022-07-25 at 17 52 44](https://user-images.githubusercontent.com/2748376/180827121-1d950f46-97a8-467e-928b-acdd98695f12.png)
